### DISCUSSION TITLE
Some notes duging testing

### DISCUSSION THREAD

**Discussion by janimattiellonen** [Bizhop/kiekkohamsteri-web#12] [General]
Adding new disc / Editing disc:
- when adding a new disc, is the photo of the disc required? Maybe the user has no image of the disc at the moment, but still wants to add the disc
- the form doesn't allow you to add a disc that your system doesn't know of
  - that might be very annoying 
- the mold and plastic fields could be an autocomplete field
  - what if the mold you were looking for isn't found?
- colour options very limited, maybe allow for own description as well?
- mold field label lacks finnish translation
- maybe add checkboxes for:
  - hole in one
  - sold
  - collection disc

Editing a disc:
  - when I clicked the edit icon (pencil) to edit a disc, the mold and plastics select lists are empty of actual options (only "Select..." and "ANY" available)
   - later saw that if I chose a manufacturer, the content of the mold changes accordingly 
  - I tried using "ergerg" as a value for price and when submitting the changes, nothing happend. WHen reviewing with the browser inspector, I could see that the PUT request failed
Disc listing page:
  - column title and column contents not aligned properly
  - does not seem to work well on smaller devices (horizontal scroll, yuck ;) )

Technical:
- using public running IDs might bite back at you at some point:
 - makes it easier for attackers to guess the next id and to try to harvest your data
 - migth be useful to consider using UUIDs for public ids while retaining numerical values for internal use (database primary key ...)

**Comment by Bizhop** [comment]
Thansk a lot Janimatti! I'll create some issues based on these comments and explain some others that are based on choices

**Comment by Bizhop** [comment]
> Adding new disc / Editing disc:
> 
> * when adding a new disc, is the photo of the disc required? Maybe the user has no image of the disc at the moment, but still wants to add the disc

This is the way the backend is designed. Frontend could send just some "no image"-image if the user doesn't want to add photo.

> * the form doesn't allow you to add a disc that your system doesn't know of
>   
>   * that might be very annoying
> * the mold and plastic fields could be an autocomplete field
>   
>   * what if the mold you were looking for isn't found?
> * colour options very limited, maybe allow for own description as well?

These are by design as well to keep data as coherent as possible. Administrators can add molds and plastics via web interface, I can promote your user so you can check that out as well.

> * mold field label lacks finnish translation

What could be the proper finnish translation? "Malli"?

> * maybe add checkboxes for:
>   
>   * hole in one
>   * sold
>   * collection disc

Hole in on and collection disc would be good. If sold via the platform the disc is actually transferred to the other user. If just sold externally, I have just removed discs from my collection.

> 
> Editing a disc:
> 
> * when I clicked the edit icon (pencil) to edit a disc, the mold and plastics select lists are empty of actual options (only "Select..." and "ANY" available)
> * later saw that if I chose a manufacturer, the content of the mold changes accordingly

This is by design, ANY is actually a row in manufacturers, molds and plastics (manufacturer "ANY" has mold "ANY" and plastic "ANY"). After manufacturer is selected you should see only molds and plastics of this manufacturer.

> * I tried using "ergerg" as a value for price and when submitting the changes, nothing happend. WHen reviewing with the browser inspector, I could see that the PUT request failed

I suck at creating forms and validations but can take a look :)

>   Disc listing page:
> * column title and column contents not aligned properly
> * does not seem to work well on smaller devices (horizontal scroll, yuck ;) )

I suck at frontend design

> 
> Technical:
> 
> * using public running IDs might bite back at you at some point:
> * makes it easier for attackers to guess the next id and to try to harvest your data
> * migth be useful to consider using UUIDs for public ids while retaining numerical values for internal use (database primary key ...)

Good idea, made tickets already to backend about this

**Reply by Bizhop** [reply]
Frontend ticket for UUID change: #13

**Comment by Bizhop** [comment]
Big changes done on both backend and frontend, name a few:
- when creating new disc, image is not required, there's initial placeholder for no-image
- making tables more compact, using sorting and pagination on most
- form validations
- disc identified by UUID
- lots of improvements under the hood

**Comment by Bizhop** [comment]
One known bug/issue is there: #17
